File: bintree_java/src/binarytreedisp/ParsingNode.java
package binarytreedisp;



class ParsingNode {
    public String key = "";
    public ParsingNode left = null;
    public ParsingNode right = null;

    public int width = 0;
    public int widthLeftBranch = 0;
    public int widthRightBranch = 0;
    public int sizeLeftLine = 0;
    public int sizeRightLine = 0;

    public int marginKey = 0;
    public int marginLeftChild = 0;
    public int marginRightChild = 0;



    public ParsingNode() {
    }



    public ParsingNode(String key) {
        this.key = key;
    }
}

File: bintree_csharp/binarytreedisp/ParsingNode.cs
namespace my.binarytreedisp
{
    class ParsingNode
    {
        public string Key = string.Empty;
        public ParsingNode Left = null;
        public ParsingNode Right = null;

        public int Width = 0;
        public int WidthLeftBranch = 0;
        public int WidthRightBranch = 0;
        public int SizeLeftLine = 0;
        public int SizeRightLine = 0;

        public int MarginKey = 0;
        public int MarginLeftChild = 0;
        public int MarginRightChild = 0;



        public ParsingNode()
        {
        }



        public ParsingNode(string key)
        {
            this.Key = key;
        }
    }
}
